src/hotspot/share/gc/serial/serialHeap.cpp

```diff
@@ -283,16 +283,12 @@ size_t SerialHeap::max_capacity() const {
 }
 
 HeapWord* SerialHeap::expand_heap_and_allocate(size_t size, bool is_tlab) {
-  HeapWord* result = nullptr;
-  if (_old_gen->should_allocate(size, is_tlab)) {
-    result = _old_gen->expand_and_allocate(size);
-  }
+  HeapWord* result = _young_gen->allocate(size);
```

Comment:
The order of `_young_gen->allocate` and `old_gen->expand_and_allocate` is reversed now. It used to be `_old_gen->expand_and_allocate == nullptr -->  _young_gen->allocate`, now it's the other way around. Is that intended?

Reply:
On master, before the method is invoked, young-gen allocation is already attempted inside `attempt_allocation(size, is_tlab, false /*first_only*/);`.

With this cleanup, it become clearer that we try young-gen first, then old-gen. So, yes, this is intended.